### DISCUSSION TITLE
useAsyncData, useFetch - support for headers?

### DISCUSSION THREAD

**Discussion by AndersNielsen85** [nuxt/nuxt#18233] [Questions] [answered]
The underlying ohmyfetch seems to support adding headers. However I can't seem to figure out how to ad them using useAsyncData/useFetch
https://github.com/unjs/ohmyfetch#-adding-headers

```
const { data: cat } = await useAsyncData('cat', () => $fetch('https://api.thecatapi.com/images/search?mime_types=gif', {
  headers: {
    'x-api-key': 'my-secret-key'
  }
}).catch((e) => {
  console.log(e)
}));
```

```
const { data: cat } = await useFetch('https://api.thecatapi.com/images/search?mime_types=gif', {
  headers: {
    'x-api-key': 'my-secret-key'
  }
});

```
I wonder if I am making some mistake, or headers are simply not supported by useAsynCData/useFetch ?

**Comment by psycongaroo** [comment]
We need an interceptors for add headers, jwt and more. ;(

**Reply by pi0** [reply]
You can create composables that intercept `useFetch` behavior:

```vue
<script setup>
const useFetchWithHeaders = (req, opts) => useFetch(req, { ...opts, headers: { test: '123' } });

const { data, refresh } = await useFetchWithHeaders('/api/test');
</script>
```

**Reply by psycongaroo** [reply]
Ty, great idea, but this is not globally.
I mean useFetch or useAsyncData use $fetch, and I should do a Composable function for both.

Is not better do a interceptors only for $fetch?

**Reply by psycongaroo** [reply]
@pi0 I missed to tag you, so sorry.

**Comment by pi0** [comment, marked as answer]
Hi @AndersNielsen85 I made a quick example but seems headers are working as expected. See this demo: https://stackblitz.com/edit/github-w4b9bd?file=app.vue

Maybe `x-api-key` does not have the correct value?

**Reply by AndersNielsen85** [reply]
Thanks a lot for the example. They are indeed working, and this made me go trough everything again.

The `x-api-key` was actually correct. However the url I provided was not(shame on me)
